include/zephyr/net/http/client.h

```diff
@@ -254,27 +254,27 @@ struct http_request {
 	size_t recv_buf_len;
 
 	/** The URL for this request, for example: /index.html */
-	const char *url;
+	char *url;
```

Comment:
I'm a bit lost here, how exactly the application is constrained by those fields being "const"? I have an impression it's the other way around, if we don't guarantee the library won't touch those fields, we couldn't assign URL constants as we do in the [HTTP client sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/sockets/http_client/src/main.c#L252) for instance.

Reply:
@rlubos my URL (with this example) is NOT constant.
At some point I have the following:
```
request.url = malloc(....)
strncpy(request.url, ...)
```

and later:
```
free(request.url);
```

The library is not touching this, but my application is.

Reply:
Actually to void the warnings at compilation time I need to do:
```
free((void *)request.url));
```

The compilation has no warning, but the static code analysis complains about discard of the const qualifier...

Reply:
I see your point, but to me it should be up to the application to either:
* Do the casting as you suggest - it's the application which sets the pointer so it knows best that it's dynamically allocated (and thus not constant) and it's safe to remove `const` qualifier in such case,
* or have a separate `char *` pointer variable to remember the allocated memory address, if you concern about the static analysis.

`const` qualifier inside `struct http_request` fields is just a promise to the application that the library won't modify the memory pointed by that field. And because of that we can safely use string constants for instance, which seams to be heavily used across the sample code given CI results.

Reply:
The second option you mentioned is not bad :-) I will check for this.
I know the interest for the const qualifier, just need to properly find the right way to work. Else not only the `char*` should be const, but all the other fields probably too ;-)